prow/cmd/deck/main.go

```diff
@@ -1031,10 +1032,13 @@ func handleLog(lc logClient) http.HandlerFunc {
 			http.Error(w, fmt.Sprintf("Log not found: %v", err), http.StatusNotFound)
 			logger := logger.WithError(err)
 			msg := "Log not found."
-			if strings.Contains(err.Error(), "PodInitializing") {
+			if strings.Contains(err.Error(), "PodInitializing") || strings.Contains(err.Error(), "not found") {
```

Comment:
also use `IsNotFound(err)` here?

Reply:
Currently this is catching both pod and prowjob not found, that would be more complicated as you'd need to do both individually.